File: leetcode/js/27_Remove Element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
    let newSize = 0;
    let end = nums.length - 1;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] == val) {
            nums[i] = nums[end];
            nums[end] = "_";
            end--;
            i--;
            newSize--;
        } else {
            newSize++;
        }
    }
    return newSize;
};

console.log(removeElement([3, 2, 2, 3], 3)); // 2
console.log(removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2)); // 5
// console.log(removeElement());
// console.log(removeElement());

File: leetcode/py/27_Remove Element.py
from typing import List
from myUtils.Utils import printResult

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        n = len(nums)
        count = 0
        for i in range(n):
            if nums[i] == val:
                count += 1
                nums[i] = 51

        nums.sort()
        return n - count

obj = Solution()

nums = [3,2,2,3]
val = 3
expected = 2
result = obj.removeElement(nums, val)
printResult(result, expected)

nums = [0,1,2,2,3,0,4,2]
val = 2
expected = 5
result = obj.removeElement(nums, val)
printResult(result, expected)

